### PR TITLE
Try a SonarQube PR issue with https://sonarqube.com/

### PR DESCRIPTION
**Pull request by axel3rd**

### CODE REVIEW

File: src/main/java/org/blondin/fa/Main.java [hunk 1]
```diff
@@ -33,6 +33,8 @@ private Main() {
      */
     public static int main(String[] args) throws IOException {
 
+        System.out.println("This is a SonarQube issue with https://sonarqube.com/");
```
Comment (axel3rd) [inline src/main/java/org/blondin/fa/Main.java:36]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS106)

**Comment by axel3rd** [comment]
SonarQube analysis reported 1 issue
- MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by trajano** [comment]
Interesting... I wonder if they added it recently after I brought it up this morning :+1:

**Comment by trajano** [comment]
I think I see the difference between yours and mine... for your case you didn't use `addon: sonarqube` I think if we had it enabled it will somehow set `sonar.scanner.skip=true` for PRs.  Need to see if that's the case

**Comment by trajano** [comment]
Confirmed I see this in the env vars, I'll ask the SonarQube guys if that was intentional or not.  Maybe we shouldn't be doing this Github pull request stuff with their public sonar.

```
SONARQUBE_SCANNER_PARAMS={ "sonar.scanner.skip" : "true" }
SONARQUBE_SKIPPED=true
```

**Comment by axel3rd** [comment]
Good catch. Perhaps could be overridden on the command line ;).
Otherwise this PR feature should come in a short time (with the travis addon, better than my "old school" config) ...

**Comment by trajano** [comment]
The question I have is whether we are allowed to do it.  Hopefully they'll say yes.  And hopefully they'll make it so that the parameters are passed in via the addon

**Comment by trajano** [comment]
Anyway there's a much cleaner way I found, but again it's still pending @SonarSource approval especially as it does not seem to be documented anywhere.

**Comment by axel3rd** [comment]
> The question I have is whether we are allowed to do it.

The PR analysis is only on the host where exectued the build => no impact on SonarQube platform. So IMO this is "tolerable" :)
But if this is not documented/supported officially, this is not sustainable and could fail at any moment without any communication.

---

Another _problem_ will be that the Maven ecosystem is built by Jenkins, so I'm not sure that the maintainer are ready using Travis (ex : [#84](https://github.com/apache/maven-plugins/pull/84)) :(

**Comment by trajano** [comment]
Okay so I guess we can go ahead with using the Github PR system sonarqube until they tell us otherwise?

**Comment by axel3rd** [comment]
> Okay so I guess we can go ahead with using the Github PR system sonarqube until they tell us otherwise?

I think yes. [Fabrice Bellingard](https://github.com/bellingard) (SonarSource Product Manager) says that it is not the best config (agree on that) but not we can't use it.

**Comment by trajano** [comment]
Alrighty, anyway I have a simplified version of all of this... no `if`s.  Here's a preview of the blog post it's still being written but the TL;DR section is done.

https://www.trajano.net/?p=1850&preview=1&_ppp=b545810f3a

**Pull request closed without merging**